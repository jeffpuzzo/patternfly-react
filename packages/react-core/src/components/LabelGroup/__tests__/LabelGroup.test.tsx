import React from 'react';

import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Label } from '../../Label';
import { LabelGroup } from '../index';

describe('LabelGroup', () => {
  test('label group default', () => {
    render(
      <LabelGroup data-testid="label-group-test-id">
        <Label>1.1</Label>
      </LabelGroup>
    );
    expect(screen.getByTestId('label-group-test-id').outerHTML).toMatchSnapshot();
  });

  test('label group with category', () => {
    render(
      <LabelGroup categoryName="category" data-testid="label-group-test-id">
        <Label>1.1</Label>
      </LabelGroup>
    );
    expect(screen.getByTestId('label-group-test-id').outerHTML).toMatchSnapshot();
  });

  test('label group with closable category', () => {
    render(
      <LabelGroup categoryName="category" isClosable data-testid="label-group-test-id">
        <Label>1.1</Label>
      </LabelGroup>
    );
    expect(screen.getByTestId('label-group-test-id').outerHTML).toMatchSnapshot();
  });

  test('label group expanded', () => {
    render(
      <LabelGroup>
        <Label>1</Label>
        <Label>2</Label>
        <Label>3</Label>
        <Label>4</Label>
      </LabelGroup>
    );
    const showMoreButton = screen.getByRole('button');

    expect(showMoreButton.textContent).toBe('1 more');

    userEvent.click(showMoreButton);
    expect(showMoreButton.textContent).toBe('Show Less');
  });

  test('label group will not render if no children passed', () => {
    render(<LabelGroup data-testid="label-group-test-id" />);
    expect(screen.queryByTestId('label-group-test-id')).toBeNull();
  });

  test('label group with category and tooltip', () => {
    render(
      <LabelGroup categoryName="A very long category name" data-testid="label-group-test-id">
        <Label>1.1</Label>
      </LabelGroup>
    );
    expect(screen.getByTestId('label-group-test-id').outerHTML).toMatchSnapshot();
  });

  test('label group compact', () => {
    render(
      <LabelGroup isCompact data-testid="label-group-test-id">
        <Label isCompact>1</Label>
        <Label isCompact>2</Label>
        <Label isCompact>3</Label>
        <Label isCompact>4</Label>
      </LabelGroup>
    );
    expect(screen.getByTestId('label-group-test-id').outerHTML).toMatchSnapshot();
  });
});
