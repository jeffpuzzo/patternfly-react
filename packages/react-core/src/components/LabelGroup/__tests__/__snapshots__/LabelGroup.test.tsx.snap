// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelGroup label group compact 1`] = `"<ul class=\\"pf-c-label-group__list\\" aria-label=\\"Label group category\\" role=\\"list\\" data-testid=\\"label-group-test-id\\"><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label pf-m-compact\\"><span class=\\"pf-c-label__content\\">1</span></span></li><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label pf-m-compact\\"><span class=\\"pf-c-label__content\\">2</span></span></li><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label pf-m-compact\\"><span class=\\"pf-c-label__content\\">3</span></span></li><li class=\\"pf-c-label-group__list-item\\"><button class=\\"pf-c-label pf-m-overflow pf-m-compact\\"><span class=\\"pf-c-label__content\\">1 more</span></button></li></ul>"`;

exports[`LabelGroup label group default 1`] = `"<ul class=\\"pf-c-label-group__list\\" aria-label=\\"Label group category\\" role=\\"list\\" data-testid=\\"label-group-test-id\\"><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label\\"><span class=\\"pf-c-label__content\\">1.1</span></span></li></ul>"`;

exports[`LabelGroup label group with category 1`] = `"<ul class=\\"pf-c-label-group__list\\" aria-labelledby=\\"pf-random-id-1\\" role=\\"list\\" data-testid=\\"label-group-test-id\\"><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label\\"><span class=\\"pf-c-label__content\\">1.1</span></span></li></ul>"`;

exports[`LabelGroup label group with category and tooltip 1`] = `"<ul class=\\"pf-c-label-group__list\\" aria-labelledby=\\"pf-random-id-4\\" role=\\"list\\" data-testid=\\"label-group-test-id\\"><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label\\"><span class=\\"pf-c-label__content\\">1.1</span></span></li></ul>"`;

exports[`LabelGroup label group with closable category 1`] = `"<ul class=\\"pf-c-label-group__list\\" aria-labelledby=\\"pf-random-id-2\\" role=\\"list\\" data-testid=\\"label-group-test-id\\"><li class=\\"pf-c-label-group__list-item\\"><span class=\\"pf-c-label\\"><span class=\\"pf-c-label__content\\">1.1</span></span></li></ul>"`;
